### PR TITLE
LibGUI: Prevent selecting empty line in TextEditor on double click

### PR DESCRIPTION
**Pull request by Sauler**

### CODE REVIEW

File: Userland/Libraries/LibGUI/TextDocument.cpp [hunk 1]
```diff
@@ -141,6 +141,18 @@ bool TextDocumentLine::ends_in_whitespace() const
     return isspace(code_points()[length() - 1]);
 }
 
+bool TextDocumentLine::can_select() const
+{
+    if (is_empty())
+        return false;
+    for (size_t i = 0; i < length(); ++i) {
+        auto code_point = code_points()[i];
+        if (isblank(code_point) || !isspace(code_point))
```
Comment (IdanHo) [inline Userland/Libraries/LibGUI/TextDocument.cpp:150]: Missing negation on isblank?

Comment (Sauler) [reply Userland/Libraries/LibGUI/TextDocument.cpp:150]: isblank checks for a blank character; that is, a space or a tab. 

I did not want to block the possibility of selecting a line containing only spaces or tabs. Just other white-space characters like newline `\n`or cariage return `\r`

Comment (IdanHo) [reply Userland/Libraries/LibGUI/TextDocument.cpp:150]: i see, thats not the most clear way to express that (wouldnt (code_point != '\n' && code_point != '\r') be easier?), but thats fine then :)

Comment (Sauler) [reply Userland/Libraries/LibGUI/TextDocument.cpp:150]: I can change it to this:
`code_point != '\n' && code_point != '\r' && code_point != '\f' && code_point != '\v'`
but i'm not sure if it's readable enough.

Another option is something like this:
`isprint(code_point) || code_point == '\t'`

Comment (IdanHo) [reply Userland/Libraries/LibGUI/TextDocument.cpp:150]: `isprint(code_point) || code_point == '\t'` is worse since that breaks unicode support :)

Comment (Sauler) [reply Userland/Libraries/LibGUI/TextDocument.cpp:150]: Ok, so i have to do this?
`code_point != '\n' && code_point != '\r' && code_point != '\f' && code_point != '\v'`

Comment (IdanHo) [reply Userland/Libraries/LibGUI/TextDocument.cpp:150]: i think thats clearer than the current check, but its up to you, if you think its fine like that, go for it :)

Comment (Sauler) [reply Userland/Libraries/LibGUI/TextDocument.cpp]: I updated it and it looks better :)

**Pull request merged**